lib/parser.js

```diff
@@ -103,6 +103,14 @@ class Parser {
       req.setTimeout(this.options.timeout, () => {
         return reject(new Error("Request timed out after " + this.options.timeout + "ms"));
       });
+      let socketTimout = this.options.timeout;
+      req.on('socket', function(socket) {
+        setTimeout(() => {
```

Comment:
What if we skip both `req.on('socket') and `req.setTimeout`, and instead handle the timeout directly?

i.e. after `req.on('error')`, just write
```js
setTimeout(() => {
  return reject(new Error("Request timed out after " + this.options.timeout + "ms"));
}, this.options.timeout);
```

Reply:
Yeah totally. Thanks for the feedback, I can make this edit and push it up sometime this evening.